ext/thymeleaf/src/main/java/org/eclipse/krazo/ext/thymeleaf/DefaultJakartaServletWebApplicationWrapperProducer.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022 Eclipse Krazo committers and contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.eclipse.krazo.ext.thymeleaf;
+
+import org.thymeleaf.web.servlet.JakartaServletWebApplication;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Inject;
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.http.HttpServletRequest;
+
+public class DefaultJakartaServletWebApplicationWrapperProducer {
+
+    @Inject
+    private HttpServletRequest request;
+    
+    @Produces
+    @ApplicationScoped
```

Comment:
Maybe we should add the `@Internal` qualifier here? We use this qualifier for some beans used internally by Krazo to protect ourselves against potential conflicts with other libraries which produce the same type as well. We use it for `HttpServletRequest` for example.

I'm not super sure if we should do it here as well. I just wanted to mention it. :-)

Reply:
I tried to understand what the `@Internal` qualifier does but I don't know if I fully get it yet. 
If using it means that the user cannot provide their own producer to override ours, then I would refrain from using it because it might be useful for the user to configure `JakartaServletWebApplication` themselves.

Also if you were afraid that the user might want to inject a `JakartaServletWebApplicationWrapper` somewhere in their code using their own producer: I don't think that this should happen as `JakartaServletWebApplicationWrapper` was written by us specifically for the purpose we are using it for and nothing else.

I hope that clarified things and that I correctly understood what `@Internal` does.